### PR TITLE
Add method to send WC profile data after profiler completion

### LINKED ISSUE #1893: Onboarding - Profile Wizard: Pass profile data back to WooCommerce.com
Per the comments in the design thread, we would like to be able to pass data via the WooCommerce helper back to WooCommerce.com. I believe this is a different need from tracks info, since we want to associate the answers with a connected site/user in marketing, etc.

This task also involves coordinating with Matty/Alpha on the WooCommerce.com side of things.

[Master Thread](https://wp.me/p90Yrv-10i) | [Designs](https://wp.me/p6riRB-4rG)

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #1893 

~~**Blocked:** The current helper API does not let us send the proper request type so this won't work until core is updated.~~

This PR sends the profiler data to wccom after the profiler is completed.

### Detailed test instructions:

1. Connect/authorize your store with WCCOM.
1.  Log the result of the newly added `WC_Helper_API::put()` response.
1. Enable the profiler and fill out all the data, completing the profiler.
1. Note the response is successful.
1. Delete `woocommerce_onboarding_profile` and revisit the profiler.
1. Skip steps by directly appending `step=theme` to the URL.
1. Note the defaults are added and profile data is still accepted.
1. Repeat these steps while disconnected from WCCOM and connect to make sure data is sent after connecting.

### CODE REVIEW

**Comment by joshuatf** [comment]
https://github.com/woocommerce/woocommerce/pull/26262 was merged, so make sure to check out the latest master of WC to test this branch.

File: src/Features/Onboarding.php [hunk 1]
```diff
@@ -141,6 +143,70 @@ private function add_filters() {
 		add_filter( 'woocommerce_show_admin_notice', array( $this, 'remove_install_notice' ), 10, 2 );
 	}
 
+	/**
+	 * Send profile data to WooCommerce.com.
+	 */
+	public static function send_profile_data() {
+		if ( ! class_exists( '\WC_Helper_API' ) && method_exists( '\WC_Helper_API', 'put' ) ) {
```
Comment (becdetat) [inline src/Features/Onboarding.php:150]: Is this check needed now that `WC_Helper_API::put` is in WooCommerce?

Comment (becdetat) [reply src/Features/Onboarding.php:150]: Oh of course it is, if wc-admin gets updated before WooCommerce. Excuse me :-)

**Comment by becdetat** [comment]
I couldn't get this to stop attempting the push, even with Jetpack and WCS both disabled.

**Comment by joshuatf** [comment]
>I couldn't get this to stop attempting the push, even with Jetpack and WCS both disabled.

@becdetat Do you mean it repeatedly attempted to send data when it could not?  I don't believe you need Jetpack and WCS installed for this to work.  There just needs to be a connection with WCCOM.

But I'm wondering now if we want to send data only if tracking has been opted in or if this is separate. /cc @timmyc @mattyza

**Comment by mattyza** [comment]
IMO, this should respect the usage tracking setting. Anything I'm missing, though? @timmyc @joshuatf

**Comment by joshuatf** [comment]
>IMO, this should respect the usage tracking setting.

I agree.  Updated in https://github.com/woocommerce/woocommerce-admin/pull/4177/commits/3329cb964d40f7fb1fa8cee147090b4d894c8f65.

**Comment by becdetat** [comment]
> Do you mean it repeatedly attempted to send data when it could not? I don't believe you need Jetpack and WCS installed for this to work. There just needs to be a connection with WCCOM.

Sorry, I was conflating the JP connection with the WCCOM connection, which I haven't had anything to do with yet. How do you disable the connection?

**Comment by joshuatf** [comment]
Thanks for the review @jeffstieler!

>A request to WCCOM was attempted even though the helper was disconnected, this resulted in an authentication error response from WCCOM

Good catch; thought I was disconnected but had to clear auth tokens from the database to get the error.

>"industry[0] is not one of 

Completely forgot that these are now formatted as individual arrays and my option still contained the old format.

Updated in https://github.com/woocommerce/woocommerce-admin/pull/4177/commits/fa2553066e61520a39d75b675663b1e926feb0df and ready for re-review.

**Comment by joshuatf** [comment]
Also as a heads up `kliken-marketing-for-google` is not an accepted param on the wccom side, but I'll open a PR there to address this.

**Comment by joshuatf** [comment]
Thanks, Jeff!  Added the newer platform options to the wccom PR.

**Pull request merged**